### ISSUE TITLE
Scrennshot validation is incorrect/inconsistent in generator-joplin / website-plugin-discovery

### ISSUE THREAD

**Issue by JackGruber** [laurent22/joplin#9649]
### Operating system

Windows

### Joplin version

generator-joplin@2.13.2

### Current behaviour

With the current implementation, as validation start path src is assumed, therefore image references only work with relative paths or if they are in src.
However, relative paths are not allowed in website-plugin-discovery and therefore no screenshots can be displayed. Even if these are in src, as the path is then incorrect for website-plugin-discovery.
https://github.com/joplin/website-plugin-discovery/blob/5ba3600d133851a44293e9a172c154edff18b79d/build/assets/PluginRemoteInfoLoader.ts#L23-L26

## Examples for manifest.json and generator-joplin

Working
```
  "screenshots": [
    {
      "src": "../img/main.png",
      "label": "Screenshot: Showing the Note liste preview with thumbnail"
    }
   ]
```

Not working (Starts validating in root)
```
  "screenshots": [
    {
      "src": "/img/main.png",
      "label": "Screenshot: Showing the Note liste preview with thumbnail"
    }
   ]
```
![grafik](https://github.com/laurent22/joplin/assets/24863925/2ce3968d-5454-448f-b539-75ccc38bf100)

Not working (Validating in src)
```
  "screenshots": [
    {
      "src": "img/main.png",
      "label": "Screenshot: Showing the Note liste preview with thumbnail"
    }
   ]
```
![grafik](https://github.com/laurent22/joplin/assets/24863925/bb91524f-6ea2-483d-bd88-909807967ad1)

Not working (Append the url to the src path)
```
  "screenshots": [
    {
      "src": "https://raw.githubusercontent.com/JackGruber/joplin-plugin-notelistpreview/master/img/main.png",
      "label": "Screenshot: Showing the Note liste preview with thumbnail"
    }
   ]
```
![grafik](https://github.com/laurent22/joplin/assets/24863925/c8f09bff-ba44-49ba-9c11-67e3f177ec50)

### Expected behaviour

- Either relative urls must be allowed in website-plugin-discovery or generator-joplin should validate from project root and not src
- generator-joplin should allow URLs

**Comment by JackGruber** [comment]
Icon is not validated from generator-joplin
```
  "icons": {
    "256": "../img/icon_256.png"
  }
```

**Comment by personalizedrefrigerator** [comment]
Thank you for reporting this!

It looks like a new version of `generator-joplin` needs to be published &mdash; validation logic was adjusted in https://github.com/laurent22/joplin/commit/1e530b74d4a7403bb9e42ead85e14fbd9ad785c6, but it doesn't seem to be in the version on NPM.


<details>

Validation logic in [`generator-joplin` on NPM](https://www.npmjs.com/package/generator-joplin?activeTab=code):
```js
function validateScreenshots(screenshots) {
	if (!screenshots) return null;
	// eslint-disable-next-line github/array-foreach -- Old code before rule was applied
	screenshots.forEach(screenshot => {
		if (!screenshot.src) throw new Error('You must specify a src for each screenshot');

		const screenshotType = screenshot.src.split('.').pop();
		if (!allPossibleScreenshotsType.includes(screenshotType)) throw new Error(`${screenshotType} is not a valid screenshot type. Valid types are: \n${allPossibleScreenshotsType}\n`);

		const screenshotPath = path.resolve(srcDir, screenshot.src);
		// Max file size is 1MB
		const fileMaxSize = 1024;
		const fileSize = fs.statSync(screenshotPath).size / 1024;
		if (fileSize > fileMaxSize) throw new Error(`Max screenshot file size is ${fileMaxSize}KB. ${screenshotPath} is ${fileSize}KB`);
	});
}
```

Expected validation logic: See https://github.com/laurent22/joplin/commit/1e530b74d4a7403bb9e42ead85e14fbd9ad785c6

</details>

**Comment by laurent22** [comment]
I believe that should be fixed in the just released v2.13.3 of the plugin generator

**Comment by JackGruber** [comment]
Yes the validation works with 2.13.3
